docs/content/development/modules/create.mdx

```diff
@@ -0,0 +1,321 @@
+    "experimentalDecorators": true,
+    "sourceMap": true,
+    "noImplicitReturns": true,
+    "strictNullChecks": true,
+    "strictFunctionTypes": true,
+    "noImplicitThis": true,
+    "allowJs": true,
+    "skipLibCheck": true,
+    "downlevelIteration": true
+  },
+  "include": ["src"],
+  "exclude": [
+    "dist",
+    "./src/**/__tests__",
+    "./src/**/__mocks__",
+    "./src/**/__fixtures__",
+    "node_modules"
+  ]
+}
+```
+
+This allows you to use the recommended TypeScript configurations and sets the output directory to `dist`. This is essential for preparing your module for testing and publishing.
+
+---
+
+## Step 4: Implement the Custom Functionality
+
+This step depends on what you’re actually implementing. For example, you can implement the cache or events module, or you can implement a commerce module. If what you’re creating has a guide, you can refer to it while implementing the functionalities.
+
+<DocCardList colSize={6} items={[
+  {
+    type: 'link',
+    href: '#',
+    label: 'Create a Cache Module',
+    customProps: {
+      icon: Icons['academic-cap-solid'],
+      description: 'Learn how to create a cache module in Medusa.',
+      isSoon: true
+    }
+  },
+  {
+    type: 'link',
+    href: '#',
+    label: 'Create an Events Module',
+    customProps: {
+      icon: Icons['academic-cap-solid'],
+      description: 'Learn how to create an events module in Medusa.',
+      isSoon: true
+    }
+  },
+]} />
+
+---
+
+### Step 5: Export Module
+
+After implementing the module, you must export a module object that helps the Medusa backend understand how to use this Module. This is done in the file `src/index.ts`.
+
+The file must export an object with the following properties:
+
+```ts
+type ModuleExports = {
+  service: Constructor<any>
+  loaders?: ModuleLoaderFunction[]
+  migrations?: any[]
+  models?: Constructor<any>[]
+  runMigrations?(
+    options: LoaderOptions,
+    moduleDeclaration: InternalModuleDeclaration
+  ): Promise<void>
+  revertMigration?(
+    options: LoaderOptions,
+    moduleDeclaration: InternalModuleDeclaration
+  ): Promise<void>
+}
+```
+
+:::note
+
+All property types such as `ModuleLoaderFunction` can be loaded from the `@medusajs/modules-sdk` package.
+
+:::
+
+Where:
+
+- `service`: This is the only required property to be exported. It should be the main service your module exposes.
+- `loaders`: (optional) an array of functions used to perform an action while loading the module. For example, you can log a message that the module has been loaded.
+- `migrations`: (optional) an array of paths to migration files that should run when the `migration` command is used with Medusa's CLI.
+- `models`: (optional) an array of entities that your module creates.
+- `runMigrations`: (optional) a function that can be used to define migrations to run when the `migration run` command is used with Medusa's CLI. The migrations will only run if they haven't already.
```

Comment:
```suggestion
- `runMigrations`: (optional) a function that can be used to define migrations to run when the `migration run` command is used with Medusa's CLI. The migrations will only run if they haven't already. This will only be executed if the module's scope is `isolated`.
```